api/app/Models/Pool.php

```diff
@@ -113,12 +114,48 @@ public function publishedPoolCandidates(): HasMany
 
     public function essentialSkills(): BelongsToMany
     {
-        return $this->belongsToMany(Skill::class, 'pools_essential_skills')->withTrashed(); // always fetch all skills for a pool
+        return $this->belongsToMany(Skill::class, 'pools_skills')
+            ->withTrashed() // pool-skills always fetches soft-deleted skill models
+            ->wherePivot('type', PoolSkillType::ESSENTIAL->name);
     }
 
     public function nonessentialSkills(): BelongsToMany
     {
-        return $this->belongsToMany(Skill::class, 'pools_nonessential_skills')->withTrashed();
+        return $this->belongsToMany(Skill::class, 'pools_skills')
+            ->withTrashed()
+            ->wherePivot('type', PoolSkillType::NONESSENTIAL->name);
+    }
+
+    /**
+     * Sync the essential skills in pools_skills
+     *
+     * @param $skills - array of skill ids
+     * @return void
+     */
+    public function setEssentialPoolSkills($skills)
```

Comment:
These functions feel like what we're trying to get away from - treating the two sets of skills differently.  I'd suggest marking them (and the two mutations in the graphql schema they support) as deprecated and not using them in tests, factories, etc.

I suppose that means you would need to add a regular skills(): BelongsToMany relationship that returns both types and can sync both types together.  That might be a bit tricky but maybe details field on experience_skill can be a good example.

Reply:
Hmm. I'm not sure if I fully agree. I think that eventually (when we want to add skill levels) we'll want to add a `poolSkill: BelongsToMany` relationship, and maybe deprecate the `essentialSkills` and `nonEssentialSkills` fields on graphql Pool type. 

But at the same time, I'm not against using mutations like this if they make writing mutations easier for us, or make using migrations easier for the client.

Reply:
I just approached with keeping things working the same per the proposed solution section, so the frontend doesn't change at all for example

> Later we can deprecate these fields in the schema in favour of a new PoolSkill type (which includes an essential/nonessential enum field) and gradually shift queries over to that.

It seemed like things might change a bunch so I attempted to maintain status quo.

Reply:
Fair enough.  Since this is blocking work this sprint I'll go along with the quicker solution for now.